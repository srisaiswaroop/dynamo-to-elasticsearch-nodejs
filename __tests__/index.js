const fs = require('fs');
const DynamoToES = require('../index');

const inputdata = fs.readFileSync(`${__dirname}/data/input.json`, 'utf8');
const input = JSON.parse(inputdata);

// get endpoint details
const { ES_ENDPOINT = 'http://localhost:9200' } = process.env;

test('transformEventToESBulk', async () => {
  const client = new DynamoToES({ node: ES_ENDPOINT });
  const result = client.transformEventToESBulk(input);
  expect(result).toMatchSnapshot();
});

test('bulkHandler', async () => {
  const client = new DynamoToES({ node: ES_ENDPOINT });
  await client.bulkHandler(input);
  // ToDO: need to finish checking for the items in the DB and remove them
});
