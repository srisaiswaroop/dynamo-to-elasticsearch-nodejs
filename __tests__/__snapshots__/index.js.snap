// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformEventToESBulk 1`] = `
Array [
  Object {
    "index": Object {
      "_id": "hashkeyuprimarykeyu",
      "_index": "tablename",
    },
  },
  Object {
    "id": "primarykeyu",
    "name": "Test U:new",
    "tenantId": "hashkeyu",
  },
  Object {
    "index": Object {
      "_id": "hashkeycprimarykeyc",
      "_index": "tablename",
    },
  },
  Object {
    "id": "primarykeyc",
    "name": "Test C:new",
    "tenantId": "hashkeyc",
  },
  Object {
    "delete": Object {
      "_id": "hashkeydprimarykeyd",
      "_index": "tablename",
    },
  },
]
`;
